sqlmesh/core/macros.py

```diff
@@ -1100,6 +1100,65 @@ def var(
     return exp.convert(evaluator.var(var_name.this, default))
 
 
+@macro()
+def deduplicate(
+    evaluator: MacroEvaluator,
+    relation: exp.Expression,
+    partition_by: t.List[exp.Expression],
+    order_by: t.List[str],
+) -> exp.Query:
+    """Returns a QUERY to deduplicate rows within a table
+
+    Args:
+        relation: table or CTE name to deduplicate
+        partition_by: column names, or expressions to use to identify a window of rows out of which to select one as the deduplicated row
+        order_by: A list of strings representing the ORDER BY clause
+
+    Example:
+        >>> from sqlglot import parse_one
+        >>> from sqlglot.schema import MappingSchema
+        >>> from sqlmesh.core.macros import MacroEvaluator
+        >>> sql = "@deduplicate(demo.table, [user_id, cast(timestamp as date)], ['timestamp desc', 'status asc'])"
+        >>> MacroEvaluator().transform(parse_one(sql)).sql()
+        'SELECT * FROM demo.table QUALIFY ROW_NUMBER() OVER (PARTITION BY user_id, CAST(timestamp AS DATE) ORDER BY timestamp DESC, status ASC) = 1'
+    """
+    if not isinstance(partition_by, list):
+        raise SQLMeshError(
+            "partition_by must be a list of columns: [<column>, cast(<column> as <type>)]"
+        )
+
+    if not isinstance(order_by, list):
+        raise SQLMeshError("order_by must be a list of strings: ['<column> <asc|desc>']")
```

Comment:
what about null ordering here? can you specify nulls first/last it in the qualify clause? if so let's maybe update the error message here

Reply:
I updated this so it's nulls first ordering by default to mirror the dbt-utils macro: https://github.com/dbt-labs/dbt-utils/blob/main/macros/sql/deduplicate.sql

this is done by default in the Ordered Class
```python
class Ordered(Expression):
    arg_types = {"this": True, "desc": False, "nulls_first": True, "with_fill": False}
```

Reply:
updated the error message